sklearn/tests/test_common.py

```diff
@@ -379,3 +381,37 @@ def test_transformer_n_iter():
 
         if hasattr(estimator, "max_iter") and name not in external_solver:
             yield check_transformer_n_iter, name, estimator
+
+
+# addresses #4467, #4465 and #4461
+# essentially tests consistency between shallow and deep
+# get_params() calls of
+# - FeatureUnion
+# - Pipeline
+def test_custom_get_params():
+    # find classes that overwrite the get_params() function
+    estimators = all_estimators(
+        include_meta_estimators=True,
+        include_other=True)
+    custom_funcs = []
+    for name, obj in estimators:
+        if (hasattr(obj, "get_params") and
+            # test if overwritten
+            obj.get_params != sklearn.base.BaseEstimator.get_params):
+            custom_funcs.append(obj)
```

Comment:
Why are you doing two loops?

I would simply do:

<pre>
if obj.get_params == ...
   continue
</pre>